res/skins/QMLDemo/Mixxx/Controls/Knob.qml

```diff
@@ -63,30 +64,51 @@ Item {
     DragHandler {
         id: dragHandler
 
-        property real valueRange: root.max - root.min
+        property real value
         property vector2d lastTranslation: Qt.vector2d(0, 0)
 
         target: null
         grabPermissions: PointerHandler.CanTakeOverFromAnything | PointerHandler.ApprovesTakeOverByAnything
-        onActiveChanged: lastTranslation = Qt.vector2d(0, 0)
+        onActiveChanged: {
+            dragHandler.value = root.value;
+            lastTranslation = Qt.vector2d(0, 0);
+        }
         onTranslationChanged: {
             const delta = lastTranslation.y - translation.y;
-            const change = valueRange * Math.max(Math.min(delta, 100), -100) / 100;
-            const value = Math.max(root.min, Math.min(root.max, root.value + change));
+            const change = (root.max - root.min) * Mixxx.MathUtils.clamp(delta, -100, 100) / 100;
+            const value = Mixxx.MathUtils.clamp(root.value + change, root.min, root.max);
```

Comment:
the first clamp seems unnecessary to me.

Reply:
PLease make a concrete suggestion. Just removing the clamp will make the value go to the minimum/maximum on the slightest mouse move.

Reply:
That doesn't seem right. I'll try to debug whats going on here then.

Reply:
I can remove both clamps and the it works just as well. Could it be a screen resolution / pixel density issue?
In addition, there is also a deadzone which makes the control ignore the drag for small movements. You would have to set  `dragThreshold: 0` but thats only been added in 5.15.

Maybe using a DragHandler is a bad idea after all...
Can you elaborate on why you are replacing the mousearea with a draghandler at all?

Reply:
If I use a mousearea, double-click to reset doesn't work anymore due to event propagation issues.

I you have a simpler, working solution, let's merge know and you open a follow up PR.

Reply:
Well IMO the knobs worked as expected previously. Or am I missing something?

Reply:
I now understand the issue. I don't think composing mouseareas together is a good idea. especially not for the purpose of being able to reset a control. IMO the better approach is to make the reset functionality part of the ControlProxy object and then just give the knob and slider a `property bool resetable: false`.

I know this is making a large part of this PR irrelevant so I can understand if you are opposed to it.
I would be ok with working on this myself though my schedule makes that difficult atm.

Reply:
The main issue is that there may be even other things you might want to do, like rightclicking support etc. Adding everything to the base componet is bad design IMHO. Trying to add config options for every single thing someone may want to use is futile and makes the code a complexity nightmare. Stuff should be flexible and composable.

Also, besides that the dragarea approach works, it also is the right thing to use semantically IMHO.

Reply:
I see your point about composability... it just seems a bit wrong how we do it. The code of the baseclass is not very maintainable like this as well. Maybe we are just missing something. I'm fine with merging as is though I'd like to look into some alternative soon.

Reply:
I mean we can also factor out the Arc feature into a separate ArcKnob if you think that helps. But other than that the Knob.qml in Mixxx.Controls is pretty slim already.

Everything that is not in the Mixxx subdirectory is skin-local and doesn't need to be reusable. We can later check which parts may be useful and move it into the Mixxx directory.

Reply:
That would make sense architectually, but there is no practical reason for doing that right now.